scripts/zkai_d128_snark_receipt_timing_setup_gate.py

```
@@ -0,0 +1,688 @@
+
+
+def timed_command(command: list[str], *, cwd: pathlib.Path | None = None, timeout_s: float = 120.0) -> float:
+    started = time.perf_counter_ns()
+    run_command(command, cwd=cwd, timeout_s=timeout_s)
+    elapsed = time.perf_counter_ns() - started
+    return elapsed / 1_000_000
+
+
+def tool_versions() -> dict[str, str]:
+    if not SNARKJS.is_file():
+        raise D128SnarkTimingSetupError("repo-local snarkjs missing; run `npm ci --prefix scripts`", layer="external_proof_tooling")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)